validator/storage_bucket_policy_only_test.rego

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPStorageBucketPolicyOnlyConstraintV1
+
+all_violations[violation] {
+	resource := data.test.fixtures.storage_bucket_policy_only.assets[_]
+	constraint := data.test.fixtures.storage_bucket_policy_only.constraints.require_bucket_policy_only
+
+	issues := deny with input.asset as resource
+		 with input.constraint as constraint
+
+	violation := issues[_]
+}
+
+# Confirm total violations count
+test_storage_bucket_policy_only_violations_count {
+	count(all_violations) == 2
```

Comment:
We're asserting there are 2 violations, but we're only test for 1 (test_storage_bucket_policy_only_violations_basic ) below? Instead of checking the count, can we add another test case for the other violation?

Reply:
updated to explicitly check the 2 violating resources